### PR TITLE
fix(label): work with virtual nodes

### PR DESCRIPTION
**Pull request by straker**
This one was a lot of changes to be able to get the rule to work with serial virtual nodes.

Part of #2183 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: lib/checks/label/implicit-evaluate.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-import { accessibleText } from '../../commons/text';
+import { closest } from '../../core/utils';
+import { accessibleTextVirtual } from '../../commons/text';
 
 function implicitEvaluate(node, options, virtualNode) {
-	const label = findUpVirtual(virtualNode, 'label');
-	if (label) {
-		return !!accessibleText(label, { inControlContext: true });
+	try {
+		const label = closest(virtualNode, 'label');
```
Comment (WilcoFiers) [inline lib/checks/label/implicit-evaluate.js:6]: This is going to create false positives. If we don't have access to the parent of `input`, we can't just decide it has no implicit label as an ancestor. I think "closest" may need to throw if it stops before it reaches a root node.

File: lib/commons/text/label-virtual.js [hunk 2]
```diff
@@ -12,28 +12,37 @@ import findUpVirtual from '../dom/find-up-virtual';
...
 		if (candidate) {
 			return candidate;
 		}
 	}
 
-	ref = findUpVirtual(node, 'label');
-	candidate = ref && visible(ref, true);
+	if (!virtualNode.actualNode) {
+		throw new TypeError('Cannot determine if non-DOM node is visible');
+	}
```
Comment (WilcoFiers) [inline lib/commons/text/label-virtual.js:42]: I'm not sure a generic "always throw" is right. Instead lets have "closest" and "visibleVirtual" decide if throwing is necessary.

File: lib/commons/text/label-virtual.js [hunk 3]
```diff
@@ -12,28 +12,37 @@ import findUpVirtual from '../dom/find-up-virtual';
...
+
+	ref = closest(virtualNode, 'label');
+	candidate = ref && visible(ref.actualNode, true);
```
Comment (WilcoFiers) [inline lib/commons/text/label-virtual.js:45]:
```suggestion
	candidate = ref && visibleVirtual(ref, true);
```

**Comment by straker** [comment]
Turns out `help-same-as-label` has been disabled for at least 2 years so no need to convert it over. Not sure why we still have it as part of the check at this point.

**Pull request merged**